Stratis.Bitcoin/Consensus/ConsensusFeature.cs

```diff
@@ -29,6 +29,7 @@ public class ConsensusFeature : FullNodeFeature
 		private readonly Signals signals;
 		private readonly ConsensusLoop consensusLoop;
 		private readonly NodeSettings nodeSettings;
+		private readonly StakeChain stakeChain;
```

Comment:
Yes. POSConsensusValidator would probably be better. I'm not familiar enough with the various implementations of PoS to know what variations would be reusable for potentially other coins. 

However, it may not be too difficult if needed in the future to extract an intermediate POW class from StratisConsensusValidator should a different POW coin be implemented.

The indicator to me that things are may be getting too complex is the increasing number of method parameters throughout, perhaps subclasses can fix some of that.